first_order_model/train.py

```diff
@@ -201,19 +202,126 @@ def train(config, generator, discriminator, kp_detector, checkpoint, log_dir, da
     generator_full = GeneratorFullModel(kp_detector, generator, discriminator, train_params)
     discriminator_full = DiscriminatorFullModel(kp_detector, generator, discriminator, train_params)
     
+    for x in dataloader:
+        if use_lr_video or use_RIFE:
+            lr_frame = F.interpolate(x['driving'], lr_size)
+            if train_params.get('encode_video_for_training', False):
+                target_bitrate = train_params.get('target_bitrate', None)
+                quantizer_level = train_params.get('quantizer_level', -1)
+                if target_bitrate == 'random':
+                    target_bitrate = np.random.randint(15, 75) * 1000
+                nr = x.get('time_base_nr', torch.ones(lr_frame.size(dim=0), dtype=int))
+                dr = x.get('time_base_dr', 30000 * torch.ones(lr_frame.size(dim=0), dtype=int))
+                x['driving_lr'] = get_frame_from_video_codec(lr_frame, nr, \
+                        dr, quantizer_level, target_bitrate) 
+            else:
+                x['driving_lr'] = lr_frame
+
+        for k in x:
+            try:
+                x[k] = x[k].cuda()
+            except:
+                pass
+
+        break
+    get_gen_input(generator_full,x)
     vgg_model = Vgg19()
     original_lpips = lpips.LPIPS(net='vgg')
     vgg_face_model = VggFace16()
+    start = total_macs(generator)
```

Comment:
Is this code backwards compatible? Should these values be computed for non-netadapt code?